### DISCUSSION TITLE
Contenerized ArchR for HPC

### DISCUSSION THREAD

**Discussion by drowsygoat** [GreenleafLab/ArchR#2195] [General]
Hello. I was wondering if there has been any update regarding a containerized version of ArchR for use on HPC clusters. I understand that it's not straightforward to make ArchR work in a containerized form, but if anything like this already exists (docker, singularity...), I would be grateful for a heads up.

**Comment by mfansler** [comment]
I am proposing that Bioconda start building ArchR (as `r-archr`). In addition to a Conda package, it also would generate a BioContainer. Any thoughts on what should be packaged are welcome on https://github.com/bioconda/bioconda-recipes/pull/51295.

**Reply by mfansler** [reply]
There are links [in this post](https://github.com/bioconda/bioconda-recipes/pull/51295#issuecomment-2407307092) with the preliminary builds for the draft recipe, including a Docker image. FYI, in case anyone wants to test.

**Reply by KyriakisDimitrios** [reply]
Hi @mfansler,

I’m running into an issue with Docker. When I try to run my container, I get an error about a missing package: "BSgenome.Hsapiens.UCSC.hg38".

Here’s the error message:
```
BSgenome for hg38 not installed! Now installing by the following:
        BiocManager::install("BSgenome.Hsapiens.UCSC.hg38")
'getOption("repos")' replaces Bioconductor standard repositories, see
'help("repositories", package = "BiocManager")' for details.
Replacement repositories:
    CRAN: https://p3m.dev/cran/__linux__/jammy/2024-10-30
Bioconductor version 3.19 (BiocManager 1.30.25), R 4.4.1 (2024-06-14)
Installing package(s) 'BSgenome.Hsapiens.UCSC.hg38'
Warning in install.packages(...) :
  'lib = "/usr/local/lib/R/site-library"' is not writable
```

Additionally, since I’m running this on an HPC node without internet access, installing the package on the fly isn’t an option.

Do you have any suggestions for pre-installing the package in the container or another workaround?

Thanks in advance!

**Reply by mfansler** [reply]
@KyriakisDimitrios tricky. If you had internet access, then you could probably specify a temporary mounted location with `R_LIBS_USER` to give R a writeable location. It might be a bit hacky to pre-populate such a directory with BSgenome.Hsapiens.UCSC.hg38.

One option might be that we build variants that include different BSgenome objects. However, I'd have to ask around Bioconda to see how this is dealt with for other tools.

If you don't want to wait for that, then you build your own container on top of the existing one and host that yourself.

**Reply by mfansler** [reply]
@KyriakisDimitrios I did [ask on Bioconda](https://github.com/bioconda/bioconda-recipes/issues/53997). One reply also suggested deriving a new container. E.g., something like

**DOCKERFILE**
```dockerfile
FROM quay.io/bioconda/bioconda-utils-build-env-cos7
RUN . /opt/conda/etc/profile.d/conda.sh && \
  conda install -y \
  r-archr==1.0.3 \
  bioconductor-bsgenome.hsapiens.ucsc.hg38==1.4.5
```
